cfme/infrastructure/resource_pool.py

```diff
@@ -195,12 +194,7 @@ def get_detail(self, *ident):
     @property
     def exists(self):
         view = navigate_to(self.parent, 'All')
-        try:
-            view.toolbar.view_selector.select('List View')
-            view.entities.paginator.find_row_on_pages(view.entities.table, name=self.name)
-            return True
-        except NoSuchElementException:
-            return False
+        return self.name in view.entities.entity_names
```

Comment:
**QUESTION**
Does `view.entities.entity_names` return entities not only from current page? Alternatively, you can check the existence like that:
```python
try:
    navigate_to(self, 'Details')
    return False
except ItemNotFound:
    return False
```

Reply:
@quarckster: while `navigate_to(self, 'Details')`, we are locating the entity and try navigate to `Detail Page`. Rather than changing page with extra navigation task. I thought its good to check the 'All' page with `view.entities.entities_name`; It will return all entities present on `All page` at that time so we can check our entity l.e. `self.name` is present in returning list or not. Its will fast and effective check for `exists()`.